### ISSUE TITLE
Order dependency of X-Forwarded-Host and X-Forwarded-Port

### ISSUE THREAD

**Issue by rhuitl** [eclipse/jetty.project#4573]
**Jetty version**
9.4.24

**Java version**
OpenJDK 11

**OS type/version**
Linux & MacOS

**Description**
I see different behavior when the `X-Forwarded-Host` and `X-Forwarded-Port` headers are reversed. The application is a Spring Boot application running either deployed as WAR on Jetty or as an executable JAR with embedded Jetty. Both variants fail the CORS check when the Port header appears before the Host header:

```
curl 'http://localhost:8080/' -H "Host: 1.2.3.4:1234" -H 'origin: https://www.example.com' -H 'X-Forwarded-Proto: https' -H 'X-Forwarded-Host: www.example.com' -H 'X-Forwarded-Port: 443'
<valid response>
```
```
curl 'http://localhost:8080/' -H "Host: 1.2.3.4:1234" -H 'origin: https://www.example.com' -H 'X-Forwarded-Proto: https' -H 'X-Forwarded-Port: 443' -H 'X-Forwarded-Host: www.example.com'
Invalid CORS request
```
I'm using `--module=http-forwarded` for WAR and `server.use-forward-headers: true` for JAR.

Use case: running the application behind an Azure Application Gateway that does not set `X-Forwarded-Host` (but `X-Forwarded-Port`). Therefore, a custom header has been added with a header rewrite rule which then appears after the port header.

**Comment by joakime** [comment]
This seems like a duplicate (or at least related to the same root cause as) Issue #4504

**Comment by rhuitl** [comment]
I saw that one and briefly looked at the code. I didn't set the `X-Forwarded-Server` header so I'm not sure that the code change would fix this issue.

**Comment by gregw** [comment]
I think I have reproduced.... stand by...

**Comment by gregw** [comment]
Ah it was a cut and paste error!   We had `_for` instead of `_host` in one test.  Fixed in 652428ad7040444d46888f652fb4a0b1a28c7832 (accidentally committed directly the main branch rather than PR).
@joakime can you review the commit and extra tests - then close if you think this is OK.

**Comment by rhuitl** [comment]
I tested it and can confirm that the issue is fixed with Jetty 9.4.27! Thank you.
